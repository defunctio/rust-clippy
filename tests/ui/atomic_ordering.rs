#![warn(clippy::atomic_ordering)]
use std::sync::atomic::{
    AtomicBool, AtomicI16, AtomicI32, AtomicI64, AtomicI8, AtomicIsize, AtomicPtr, AtomicU16, AtomicU32, AtomicU64,
    AtomicU8, AtomicUsize, Ordering,
};

macro_rules! order_test {
    ( [ $($typ:ty),+ ], $op:ident, $ord:expr ) => (
        $(
            let g = <$typ>::new(1);
            g.$op(10, $ord);
        )+
    );
    ( $typ:ty, $op:ident, $ord:expr, $val:expr ) => {
            let g = <$typ>::new($val);
            g.$op($val, $ord);
    };
}

fn main() {
    // Test invalid signed/unsigned
    order_test!(
        [AtomicI16, AtomicI32, AtomicI64, AtomicI8, AtomicIsize],
        store,
        Ordering::Acquire
    );
    order_test!(
        [AtomicI16, AtomicI32, AtomicI64, AtomicI8, AtomicIsize],
        store,
        Ordering::AcqRel
    );
    order_test!(
        [AtomicU16, AtomicU32, AtomicU64, AtomicU8, AtomicUsize],
        store,
        Ordering::Acquire
    );
    order_test!(
        [AtomicU16, AtomicU32, AtomicU64, AtomicU8, AtomicUsize],
        store,
        Ordering::AcqRel
    );

    //AtomicBool test
    order_test!(AtomicBool, store, Ordering::Acquire, true);
    order_test!(AtomicBool, store, Ordering::AcqRel, true);

    //AtomicPtr test
}
