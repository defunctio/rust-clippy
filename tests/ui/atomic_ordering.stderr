error: `Ordering::Acquire` is not a valid `atomic::Ordering` for `store`
  --> $DIR/atomic_ordering.rs:25:9
   |
LL |         Ordering::Acquire
   |         ^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::atomic-ordering` implied by `-D warnings`
   = help: valid orderings are ["SeqCst", "Release", "Relaxed"]

error: `Ordering::Acquire` is not a valid `atomic::Ordering` for `store`
  --> $DIR/atomic_ordering.rs:25:9
   |
LL |         Ordering::Acquire
   |         ^^^^^^^^^^^^^^^^^
   |
   = help: valid orderings are ["SeqCst", "Release", "Relaxed"]

error: `Ordering::AcqRel` is not a valid `atomic::Ordering` for `store`
  --> $DIR/atomic_ordering.rs:30:9
   |
LL |         Ordering::AcqRel
   |         ^^^^^^^^^^^^^^^^
   |
   = help: valid orderings are ["SeqCst", "Release", "Relaxed"]

error: `Ordering::Acquire` is not a valid `atomic::Ordering` for `store`
  --> $DIR/atomic_ordering.rs:35:9
   |
LL |         Ordering::Acquire
   |         ^^^^^^^^^^^^^^^^^
   |
   = help: valid orderings are ["SeqCst", "Release", "Relaxed"]

error: `Ordering::AcqRel` is not a valid `atomic::Ordering` for `store`
  --> $DIR/atomic_ordering.rs:40:9
   |
LL |         Ordering::AcqRel
   |         ^^^^^^^^^^^^^^^^
   |
   = help: valid orderings are ["SeqCst", "Release", "Relaxed"]

error: `Ordering::Acquire` is not a valid `atomic::Ordering` for `store`
  --> $DIR/atomic_ordering.rs:44:36
   |
LL |     order_test!(AtomicBool, store, Ordering::Acquire, true);
   |                                    ^^^^^^^^^^^^^^^^^
   |
   = help: valid orderings are ["SeqCst", "Release", "Relaxed"]

error: `Ordering::AcqRel` is not a valid `atomic::Ordering` for `store`
  --> $DIR/atomic_ordering.rs:45:36
   |
LL |     order_test!(AtomicBool, store, Ordering::AcqRel, true);
   |                                    ^^^^^^^^^^^^^^^^
   |
   = help: valid orderings are ["SeqCst", "Release", "Relaxed"]

error: aborting due to 7 previous errors

